testplan/testing/multitest/base.py

```diff
@@ -38,22 +40,49 @@
 def iterable_suites(obj):
     """Create an iterable suites object."""
     suites = [obj] if not isinstance(obj, collections.Iterable) else obj
+    for suite in suites:
+        suite.__name__ = suite.__class__.__name__
+        if "name" in suite.__dict__:
+            raise AttributeError(
+                "Attribute `name` can only be added into a test suite object"
+                " by using argument `name` in @testsuite decorator"
+            )
```

Comment:
because if user set ``suite.name =  xxx`` before add it into multiTest, then we have no chance to verify this attribute, the first time get_testsuite_name is called the name attribute is checked and saved, and it will not change even some people pass a function like ``lambda cls_name, suite: str(time.time())``, also it it time saving. So there' should be such a name attribute defined by user but not added by @testsuite(name=...) syntax.